src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -41,12 +42,27 @@ internal class UnixPwmChannel : PwmChannel
             Validate();
             Open();
 
-            // avoid opening the file for operations changing relatively frequently
-            _dutyCycleWriter = new StreamWriter(new FileStream($"{_channelPath}/duty_cycle", FileMode.Open, FileAccess.ReadWrite));
-            _frequencyWriter = new StreamWriter(new FileStream($"{_channelPath}/period", FileMode.Open, FileAccess.ReadWrite));
+            var timeout = OpenTimeout;
+            // If we're running as non-root (and the udev rule needs to kick in here) we 
+            // might need a few retries until we can open the channel files. 
+            while (timeout > TimeSpan.Zero)
+            {
+                try
+                {
+                    // avoid opening the file for operations changing relatively frequently
+                    _dutyCycleWriter = new StreamWriter(new FileStream($"{_channelPath}/duty_cycle", FileMode.Open, FileAccess.ReadWrite));
+                    _frequencyWriter = new StreamWriter(new FileStream($"{_channelPath}/period", FileMode.Open, FileAccess.ReadWrite));
+                    break;
+                }
+                catch (Exception x) when (x is UnauthorizedAccessException || x is IOException)
+                {
+                    Thread.Sleep(1);
+                    timeout -= TimeSpan.FromMilliseconds(1);
+                }
+            }
 
             SetFrequency(frequency);
-            DutyCycle = dutyCycle;
+            SetDutyCycle(dutyCycle);
```

Comment:
These two calls are essentially the same thing, so I don't see much value but it does make sense to keep consistency and we are calling SetFrequency so I'm fine with that.

Reply:
That's why I changed it.